### ISSUE TITLE
[YouTube Search API] Possibilty of paginate results with filter

### ISSUE THREAD

**Issue by hilmanski** [serpapi/public-roadmap#1421]
Based on the documentation page, we can use `sp` parameter both for providing `next_page_token` and `filter` results. 
https://serpapi.com/youtube-search-api

What if users want to add both pieces of information?
In case this is not possible, we should clarify it in the documentation.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000059348?view=List)

**Comment by aciddjus** [comment]
`next_page_token` already contains the filter data set on the first page. The user does not need to set it again.

Steps:
1. Make a search with `search_query` and `sp` (filters you want to set)
2. Use `next_page_token` to paginate with the initial filters applied

Example:
- 1st page [search](https://serpapi.com/searches/65d334e933b236334d020619/inspect) where I have applied the next `sp` filters: `type: video, duration: over 20 minutes, features: 4K`
- 2nd page [search](https://serpapi.com/searches/65d3358dd5b1c5a4c313078d/inspect) where I have updated my `sp` parameter with the `next_page_token` from the previous search, keeping all filters data
